File: code/jvm/repository-jdbi/src/main/java/pt/isel/repositories/jdbi/utils/JdbiConfig.java
package pt.isel.repositories.jdbi.utils;

import org.jdbi.v3.core.Jdbi;
import org.jdbi.v3.postgres.PostgresPlugin;
import pt.isel.domain.security.PasswordValidationInfo;

public class JdbiConfig {
    public static Jdbi configureWithAppRequirements(Jdbi jdbi) {
        jdbi.installPlugin(new PostgresPlugin());
        jdbi.registerColumnMapper(PasswordValidationInfo.class, new PasswordValidationInfoMapper());
        return jdbi;
    }
}

File: code/jvm/repository-jdbi/src/main/kotlin/pt/isel/repositories/jdbi/JdbiConfig.kt
package pt.isel.repositories.jdbi

import org.jdbi.v3.core.Jdbi
import org.jdbi.v3.core.kotlin.KotlinPlugin
import org.jdbi.v3.postgres.PostgresPlugin
import pt.isel.domain.security.PasswordValidationInfo
import pt.isel.repositories.jdbi.mapper.InstantMapper
import pt.isel.repositories.jdbi.mapper.PasswordValidationInfoMapper
import java.time.Instant

fun Jdbi.configureWithAppRequirements(): Jdbi {
    installPlugin(KotlinPlugin())
    installPlugin(PostgresPlugin())
    registerColumnMapper(PasswordValidationInfo::class.java, PasswordValidationInfoMapper())
    registerColumnMapper(Instant::class.java, InstantMapper())
    return this
}
